docs/v3/core/index.rst

```diff
@@ -1598,38 +1520,125 @@ Storage transformers may be stacked to combine different functionalities:
 
 .. _extensions_section:
 
-Extension points
-================
+Extensions
+==========
 
+This section describes how additional functionality can defined
+for Zarr datasets by extended the `metadata documents`_.
+
+Extension points
+----------------
 
 Different types of extensions can exist and they can be grouped as follows:
 
-=========== ======================= ================================================
-level       extension               metadata
-=========== ======================= ================================================
-array       data type               `data_type`_
-array       chunk grid              `chunk_grid`_
-array       chunk key encoding      `chunk_key_encoding`_
-array       codecs                  `codecs`_
-array       storage transformer     `storage_transformers`_
-=========== ======================= ================================================
+=========== ======================= ====================================== ================================
+level       extension               metadata                               core definitions
+=========== ======================= ====================================== ================================
+array       data type               `array-metadata-data-type`_              :ref:`data-types-list`
+array       chunk grid              `array-metadata-chunk-grid`_             :ref:`chunk-grid-list`
+array       chunk key encoding      `array-metadata-chunk-key-encoding`_     :ref:`chunk-key-encoding-list`
+array       codecs                  `array-metadata-codecs`_                 :ref:`codecs-list`
+array       storage transformer     `array-metadata-storage-transformers`_   :ref:`storage-transformers-list`
+=========== ======================= ====================================== ================================
 
 If such extension points are used by groups or arrays, they are required.
```

Comment:
old text but what does this sentence mean?

Reply:
This is perhaps the (terse) explanation of your point in https://github.com/zarr-developers/zarr-specs/pull/330/files#r1962198166 -- i.e. use of these extension points falls under `must_understand=True`. I'll look into unifying.